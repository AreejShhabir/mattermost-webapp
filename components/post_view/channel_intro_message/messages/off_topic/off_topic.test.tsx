// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {archivedChannel, offTopicChannel} from '../test_utils';

import OffTopicIntroMessage from './off_topic';

const component = (otherProps: any) => (
    <OffTopicIntroMessage
        channel={offTopicChannel}
        usersLimit={10}
        {...otherProps}
    />
);

describe('components/post_view/ChannelIntroMessages', () => {
    describe('test OFF_TOPIC Channel', () => {
        test('should match snapshot, show board button', () => {
            expect(shallow(component({}))).toMatchSnapshot();
        });

        test('should match snapshot, do not show board button', () => {
            expect(shallow(component({
                channel: archivedChannel,
            }))).toMatchSnapshot();
        });
    });
});
