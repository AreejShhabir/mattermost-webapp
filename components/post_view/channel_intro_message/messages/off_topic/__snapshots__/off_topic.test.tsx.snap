// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test OFF_TOPIC Channel should match snapshot, with boards 1`] = `
<Fragment>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <MemoizedFormattedMessage
      defaultMessage="This is the start of {display_name}, a channel for non-work-related conversations."
      id="intro_messages.offTopic"
      values={
        Object {
          "display_name": "test channel",
        }
      }
    />
  </p>
  <Memo(AddMembersButton)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "off-topic",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
    createBoard={<Memo(Connect(Component)) />}
    showSetHeader={true}
    usersLimit={10}
  />
</Fragment>
`;

exports[`components/post_view/ChannelIntroMessages test OFF_TOPIC Channel should match snapshot, without boards 1`] = `
<Fragment>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <MemoizedFormattedMessage
      defaultMessage="This is the start of {display_name}, a channel for non-work-related conversations."
      id="intro_messages.offTopic"
      values={
        Object {
          "display_name": "test channel",
        }
      }
    />
  </p>
  <Memo(AddMembersButton)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "off-topic",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
    createBoard={<Memo(Connect(Component)) />}
    showSetHeader={true}
    usersLimit={10}
  />
</Fragment>
`;
